src/types/media-session.ts

```diff
@@ -89,6 +89,28 @@ export class GenesysCloudMediaSession extends MediaSession {
     }
   }
 
+  end (reason: JingleReasonCondition | JingleReason = 'success', silent = false) {
+    this.state = 'ended';
+    this.processingQueue.kill();
+
+    if (typeof reason === 'string') {
+      reason = { condition: reason };
+    }
+
+    if (!silent) {
+      this.send('session-terminate', { reason });
+    }
+
+    setTimeout(() => {
+      if (this.pc) {
```

Comment:
I think this case will always be true. We should check the connection status: https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/connectionState.

We probably care if connectionState is `connecting` or `connected`

Reply:
Can you also add a few tests for this please?